csharp/src/Ice/WSTransceiver.cs

```diff
@@ -10,215 +10,203 @@
 using System.Runtime.InteropServices;
 using System.Security.Cryptography;
 using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
 
 namespace ZeroC.Ice
 {
-    // TODO: refactor and eliminate these exceptions
-    public sealed class WSProtocolException : Exception
+    internal sealed class WSTransceiver : ITransceiver
     {
-        internal WSProtocolException(string message)
-            : base(message)
-        {
-        }
-    }
+        public Socket? Socket => _underlying.Socket;
+        internal IReadOnlyDictionary<string, string> Headers => _parser.GetHeaders();
+        internal SslStream? SslStream => (_underlying as SslTransceiver)?.SslStream;
 
-    public sealed class WSMemoryLimitException : Exception
-    {
-        internal WSMemoryLimitException()
+        private enum OpCode : byte
         {
-        }
-    }
+            Continuation = 0x0,
+            Text = 0x1,
+            Data = 0x2,
+            Close = 0x8,
+            Ping = 0x9,
+            Pong = 0xA
+        };
+
+        private enum ClosureStatusCode : short
+        {
+            Normal = 1000,
+            Shutdown = 1001
+        };
 
-    internal sealed class WSTransceiver : ITransceiver
-    {
-        internal SslStream? SslStream => (_delegate as SslTransceiver)?.SslStream;
+        private const byte FLAG_FINAL = 0x80;   // Last frame
```

Comment:
The const names should be PascalCase. And maybe these const should be replaced by an enum.